tests/runtest.sh

```diff
@@ -22,120 +53,597 @@ function print_results {
+    line="  "
+    line="${line}<assembly"
+    line="${line} name=\"CoreClrTestAssembly\""
+    line="${line} total=\"${countTotalTests}\""
+    line="${line} passed=\"${countPassedTests}\""
+    line="${line} failed=\"${countFailedTests}\""
+    line="${line} skipped=\"${countSkippedTests}\""
+    line="${line} errors=\"${errorCount}\""
+    line="${line}>"
+    echo "$line" >>"$xunitOutputPath"
+
+    # <collection ...>
+    line="    "
+    line="${line}<collection"
+    line="${line} name=\"CoreClrTestCollection\""
+    line="${line} total=\"${countTotalTests}\""
+    line="${line} passed=\"${countPassedTests}\""
+    line="${line} failed=\"${countFailedTests}\""
+    line="${line} skipped=\"${countSkippedTests}\""
+    line="${line}>"
+    echo "$line" >>"$xunitOutputPath"
+
+    # <test .../> <test .../> ...
+    if [ -f "$xunitTestOutputPath" ]; then
+        cat "$xunitTestOutputPath" >>"$xunitOutputPath"
+        rm -f "$xunitTestOutputPath"
+    fi
+
+    # </collection>
+    line="    "
+    line="${line}</collection>"
+    echo "$line" >>"$xunitOutputPath"
+
+    if [ -n "$errorSource" ]; then
+        # <errors>
+        line="    "
+        line="${line}<errors>"
+        echo "$line" >>"$xunitOutputPath"
+
+        # <error ...>
+        line="      "
+        line="${line}<error"
+        line="${line} type=\"TestHarnessError\""
+        line="${line} name=\"${errorSource}\""
+        line="${line}>"
+        echo "$line" >>"$xunitOutputPath"
+
+        # <failure .../>
+        line="        "
+        line="${line}<failure>${errorMessage}</failure>"
+        echo "$line" >>"$xunitOutputPath"
+
+        # </error>
+        line="      "
+        line="${line}</error>"
+        echo "$line" >>"$xunitOutputPath"
+
+        # </errors>
+        line="    "
+        line="${line}</errors>"
+        echo "$line" >>"$xunitOutputPath"
+    fi
+
+    # </assembly>
+    line="  "
+    line="${line}</assembly>"
+    echo "$line" >>"$xunitOutputPath"
+
+    # </assemblies>
+    echo '</assemblies>' >>"$xunitOutputPath"
+}
+
+function exit_with_error {
+    local errorSource=$1
+    local errorMessage=$2
+    local printUsage=$3
+
+    if [ -z "$printUsage" ]; then
+        ((printUsage = 0))
+    fi
+
+    echo "$errorMessage"
+    xunit_output_end "$errorSource" "$errorMessage"
+    if ((printUsage != 0)); then
+        print_usage
+    fi
+    exit 1
+}
+
 # Handle Ctrl-C. We will stop execution and print the results that
 # we gathered so far.
 function handle_ctrl_c {
+    local errorSource='handle_ctrl_c'
+
     echo ""
     echo "*** Stopping... ***"
     print_results
-    exit 0
+    exit_with_error "$errorSource" "Test run aborted by Ctrl+C."
 }
 
 # Register the Ctrl-C handler
 trap handle_ctrl_c INT
 
+function set_core_root {
+    # Environment variables in Windows are case-insensitive. Account for the difference by allowing some casing combinations.
+    export core_root=$1
+    export Core_Root=$1
+    export CORE_ROOT=$1
```

Comment:
The .sh scripts and tests are already normalized to use Core_Root, I had found only one issue and fixed that.

The msbuild targets though, use CORE_ROOT and core_root in many places. I didn't want to change all of those at this point as it is out of scope of this change. Perhaps when then test build is enabled outside Windows, it would make sense to normalize the remaining cases as well.

For now I figured it doesn't hurt to support just three variants, as failures due to that are not very interesting. But I agree it would be nice and clean to normalize this eventually.